specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-03-01-preview/costallocation.json

```diff
@@ -0,0 +1,689 @@
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Response of cost allocation rule name availability.",
+            "schema": {
+              "$ref": "#/definitions/CostAllocationRuleCheckNameAvailabilityResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.CostManagement/costAllocationRules/{ruleName}": {
+      "get": {
+        "tags": [
+          "CostAllocationRules"
+        ],
+        "operationId": "CostAllocationRules_Get",
+        "description": "Get a cost allocation rule by rule name and billing account or enterprise enrollment.",
+        "x-ms-examples": {
+          "CostAllocationRules": {
+            "$ref": "./examples/CostAllocationRuleGet.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/billingAccountIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ruleNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/CostAllocationRuleDefinitionResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "CostAllocationRules"
+        ],
+        "operationId": "CostAllocationRules_CreateOrUpdate",
+        "description": "Create/Update a rule to allocate cost between different resources within a billing account or enterprise enrollment.",
+        "x-ms-examples": {
+          "CostAllocationRulesCreateResourceGroup": {
+            "$ref": "./examples/CostAllocationRuleCreate.json"
+          },
+          "CostAllocationRulesCreateTag": {
+            "$ref": "./examples/CostAllocationRuleCreateTag.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/billingAccountIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ruleNameParameter"
+          },
+          {
+            "name": "costAllocationRule",
+            "description": "Cost allocation rule to be created or updated",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CostAllocationRuleDefinitionRequest"
```

Comment:
the request body and response body must be same. Any properties which cant be set in the request but are returned in response can be marked as readonly. It will help in GET-PUT pipeline where customer can use the sme model